sympy/matrices/tests/test_matrices.py

```diff
@@ -2082,6 +2084,9 @@ def test_row_insert():
         l = [1, 0, 0]
         l.insert(i, 4)
         assert flatten(eye(3).row_insert(i, r4).col(0).tolist()) == l
+    a = Matrix([1, 2])
+    assert Matrix([]).row_insert(10, a) == a
```

Comment:
I think for safety reasons, this should raise an error. If someone thinks they are working with a matrix that has size and they aren't, the error should not pass silently. Though a case could be made that list slices that contain out of range bounds are silently truncated as in `L = [0, 1, 2]; L[:10] = [42]` -> `L == [42]`. Do you see any merit in either approach?